### PR TITLE
Dotnet 8

### PR DESCRIPTION
**Pull request by jtkech**
Related to #13428 and https://github.com/dotnet/razor/issues/8399

For now, using the available dotnet 8 preview 6 packages don't break the Razor Compilation, but not sure this will not be the case soon, maybe we would need to use some microsoft dev feeds.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GithubHandler.cs [hunk 1]
```diff
@@ -16,9 +16,16 @@ namespace OrchardCore.GitHub.Configuration
 {
     public class GitHubHandler : OAuthHandler<GitHubOptions>
     {
+
+#if !NET8_0_OR_GREATER
         public GitHubHandler(IOptionsMonitor<GitHubOptions> options, ILoggerFactory logger, UrlEncoder encoder, ISystemClock clock)
             : base(options, logger, encoder, clock)
         { }
+#else
+        public GitHubHandler(IOptionsMonitor<GitHubOptions> options, ILoggerFactory logger, UrlEncoder encoder)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GithubHandler.cs:25]: If this also works for NET7.0 then we should use this in all cases.

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GithubHandler.cs:25]: Doesn’t work on net7.0

**Comment by sebastienros** [comment]
Starting with `8.0-rc1` `AddRouting()` resolves `IMeterFactory` which is a new service that needs to be registered: https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs#L92

These lines will need to be added:
https://github.com/dotnet/aspnetcore/blob/445e2cf36afc3129e0626edaf8bcbf525a6d188e/src/Hosting/Hosting/src/WebHostBuilder.cs#L297-L298

**Comment by jtkech** [comment]
Okay,

`.AddRouting()` now registers a new `RoutingMetrics` singleton

https://github.com/dotnet/aspnetcore/blob/445e2cf36afc3129e0626edaf8bcbf525a6d188e/src/Http/Routing/src/DependencyInjection/RoutingServiceCollectionExtensions.cs#L111

So because we isolate, not only, but routing singletons, we will already have one per tenant.

Then `RoutingMetrics` use `IMeterFactory`, maybe okay to share the factory but I think there will be a problem to create meters multiple times with the same name, I will look at the code of `DefaultMeterFactory`.

Then maybe no problem to just let the registration of `HostingMetrics` from the host.

I will look at it.

**Comment by sebastienros** [comment]
My concern was only that the application fails on all requests. Note that there is a PR right now that will fix the issue (I reported it) and RC1 should just work. So discussion closed.

**Comment by jtkech** [comment]
Okay

**Comment by MikeAlhayek** [comment]
Maybe we should remove .NET 6 here. https://github.com/OrchardCMS/OrchardCore/pull/14234#issuecomment-1698223532

**Comment by sebastienros** [comment]
@MikeAlhayek sure, since it will only ship with net8.0. We could also remove net7.0 since 8.0 is LTS, but I'd rather do it in a next step so there is some time to do a migration from 7.0 to 8.0 under the same OC version. So 1.8 removes net6.0 and 1.9 removes net7.0.

**Comment by jtkech** [comment]
Yes I already saw the related comment, just waiting for a confirmation

**Comment by jtkech** [comment]
Okay, the confirmation has come before my previous comment ;)

**Review by sebastienros — APPROVED** [review]

**Comment by sebastienros** [comment]
I'd say let's merge it since we don't expect to release 1.8 until net8.0 is released.

**Comment by sebastienros** [comment]
Or add net6.0 back as a supported one until we decide to drop it in a future release (not test, just building and allowing the net6.0 tfm).

**Comment by jtkech** [comment]
Okay, let me know what you prefer, hmm I saw that the dotnet 8.0 release is planned to be shipped in around 3 or 4 weeks.

**Comment by MikeAlhayek** [comment]
@sebastienros @jtkech 
Personally, I think it's a good idea to merge this PR with .NET 6 support just so we can see if there is a performance gain coming with .NET8 that impacts OC. Once .NET 8 is release (November 14th), we can submit another PR to remove .NET 6 support. When we ship 1.8, we'll ship it with .NET 7+ support.

**Comment by jtkech** [comment]
No problem, just let me know

**Comment by MikeAlhayek** [comment]
> Or add net6.0 back as a supported one until we decide to drop it in a future release (not test, just building and allowing the net6.0 tfm).

@jtkech I think @sebastienros is saying to keep .NET 6 support "for now" and merge this one.

**Pull request merged**